README.md

```diff
@@ -22,50 +22,118 @@ although basic usage is shown below.
 
 ### User Registration Flow
 
+##### Server endpoint (registration challenge):
+
 ```javascript
-// 1. Check that the user is logged in.
+const u2f = require('u2f');
 
-// 2. (On server) generate user registration request and save it in session:
-var u2f = require('u2f');
+function handler(req, res) {
+  // 1. Check that the user is logged in.
 
-var req = u2f.request("<appId>");
-session.authRequest = req;
+  // 2. Generate a registration request and save it in the session.
+  // The app ID is usually the fully qualified URL of your website. The website MUST be HTTPS, or
+  // else the registration will fail client-side.
+  const registrationReq = u2f.request('<appId>');
+  req.session.registrationReq = registrationReq;
 
-// 3. Send req to the client and use U2F API to request registration, then send result back.
-// (On client)  u2f.register([req], [], function(res) { /* send res back to server */ })
+  // 3. Send the registration request to the client, who will use the Javascript U2F API to sign
+  // the registration request, and send it back to the server for verification.
+  return res.send(registrationReq);
+}
+```
 
-// 4. (Server) Check registration result.
-var checkres = u2f.checkRegistration(session.authRequest, res);
+##### Server endpoint (registration verification):
 
-if (checkres.successful) {
-    // Registration successful, save 
-    // checkres.keyHandle and checkres.publicKey to user's account in your db.
-} else {
-    // checkres.errorMessage will contain error text.
+```javascript
+const u2f = require('u2f');
+
+function handler(req, res) {
+  // 4. Verify the registration response from the client against the registration request saved
+  // in the server-side session.
+  const result = u2f.checkRegistration(req.session.registrationReq, req.body.registrationRes);
+
+  if (result.successful) {
+    // Success!
+    // Save result.publicKey and result.keyHandle to the server-side datastore, associated with
+    // this user.
+    return res.sendStatus(200);
+  }
+
+  // result.errorMessage is defined with an English-language description of the error.
+  return res.send({result});
 }
 ```
 
+##### Client logic:
+
+Note that the `window.u2f` object is defined in the official [Javascript U2F API](https://github.com/google/u2f-ref-code), for which a polyfill is [available as an npm module](https://www.npmjs.com/package/u2f-api-polyfill).
+
+```javascript
+const registrationReq = ...  // Retrieve this from hitting the registration challenge endpoint
+const {appId, version, challenge} = registrationReq;
+
+window.u2f.register(appId, [{version, challenge}], [], (registrationRes) => {
```

Comment:
The input parameters to `u2f.register` are fixed here relative to what's already in the README. Reference: [official Javascript U2F client API implementation](https://demo.yubico.com/js/u2f-api.js) and [FIDO API spec](https://fidoalliance.org/specs/fido-u2f-v1.0-nfc-bt-amendment-20150514/fido-u2f-javascript-api.html)

Relevant snippet:

```
/**
 * Dispatches register requests to available U2F tokens. An array of sign
 * requests identifies already registered tokens.
 * If the JS API version supported by the extension is unknown, it first sends a
 * message to the extension to find out the supported API version and then it sends
 * the register request.
 * @param {string=} appId
 * @param {Array<u2f.RegisterRequest>} registerRequests
 * @param {Array<u2f.RegisteredKey>} registeredKeys
 * @param {function((u2f.Error|u2f.RegisterResponse))} callback
 * @param {number=} opt_timeoutSeconds
 */
u2f.register = function(appId, registerRequests, registeredKeys, callback, opt_timeoutSeconds) {
```